### PR TITLE
support view related privileges in mysql.tables_priv

### PR DESCRIPTION
**Pull request by erjiaqing**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/9893

mysql.tables_priv does not support view related privileges

### What is changed and how it works?

In `bootstrap.go`, add view related privileges to table definition.

### Check List 

Tests 

 - **More tests is needed, but how can I simulate another user in explain test or some place else?**
 - Unit test
 - Manual test

```sql
use test;
create table t ( c int );
create view v as select * from t;
create user u@'%';
grant select, show view on v to u@'%';
grant create view on vv to u@'%';
```

```sql
use test;
show create view v;
create view vv as select * from v; 
```

```
➜  ~ mysql -h 127.0.0.1 -u root -P 4000 -p
mysql> use test;
Database changed
mysql> create table t ( c int );
Query OK, 0 rows affected (0.01 sec)

mysql> create view v as select * from t;
Query OK, 0 rows affected (0.00 sec)

mysql> create user u@'%';
Query OK, 1 row affected (0.00 sec)

mysql> grant select, show view on v to u@'%';
Query OK, 0 rows affected (0.01 sec)

mysql> grant create view on vv to u@'%';
Query OK, 0 rows affected (0.01 sec)
```

```
➜  ~ mysql -h 127.0.0.1 -u u -P 4000 -p
mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show create view v;
+------+--------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View | Create View                                                                                                  | character_set_client | collation_connection |
+------+--------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v    | CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`127.0.0.1` SQL SECURITY DEFINER VIEW `v` (`c`) AS select * from t | utf8                 | utf8_general_ci      |
+------+--------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.00 sec)

mysql> create view vv as select * from v;
Query OK, 0 rows affected (0.01 sec)

mysql>
```

Code changes

 - Has exported function/method change
> grant show view, create view is supported.

 - Has persistent data change
> mysql.tables_priv is changed
> currentBootstrapVersion set to 29

### CODE REVIEW

**Comment by wjhuang2016** [comment]
Maybe you can modify SessionVars.User to simulate another user

**Comment by wjhuang2016** [comment]
Could you please fix the other privs? such as `drop role` `show databases`

**Comment by erjiaqing** [comment]
> Could you please fix the other privs? such as `drop role` `show databases`

I found that in mysql.user, we have already supports these two privileges, so maybe we can create another pr for such commands

File: session/session_test.go [hunk 1]
```diff
@@ -2490,3 +2490,14 @@ func (s *testSessionSuite) TestTxnGoString(c *C) {
...
+
+func (s *testSessionSuite) TestGrantViewRelated(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create table if not exists t (a int)")
+	tk.MustExec("create view v_version29 as select * from t")
+	tk.MustExec("create user 'u_version29'@'%'")
+
+	// Test grant view related privileges.
+	tk.MustExec(`grant show view on v_version29 to 'u_version29'@'%'`)
+	tk.MustExec(`grant create view on v_version29_c to 'u_version29'@'%'`)
```
Comment (eurekaka) [inline session/session_test.go]: Select `mysql.tables_priv` to check the privilege is granted?

File: session/bootstrap.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ const (
 		Table_name	CHAR(64),
 		Grantor		CHAR(77),
 		Timestamp	Timestamp DEFAULT CURRENT_TIMESTAMP,
-		Table_priv	SET('Select','Insert','Update','Delete','Create','Drop','Grant', 'Index','Alter'),
+		Table_priv	SET('Select','Insert','Update','Delete','Create','Drop','Grant','Index','Alter','Create View','Show View','Trigger','References'),
```
[the remark this answers is not in the document]
Comment (erjiaqing) [reply session/bootstrap.go]: I think we can just add them in case.

**Comment by eurekaka** [comment]
Have you tried this suggestion for testing?
> Maybe you can modify SessionVars.User to simulate another user

Does it work?

**Comment by erjiaqing** [comment]
I tried `tk.Se.GetSessionVars()`, but got panic.

```go
func (s *testSessionSuite) TestGrantViewRelated(c *C) {
	tk := testkit.NewTestKitWithInit(c, s.store)
	tk.MustExec("create table if not exists t (a int)")
	tk.MustExec("create view v_version29 as select * from t")
	tk.MustExec("create user 'u_version29'@'%'")

	origUser := tk.Se.GetSessionVars().User
```

`tk.Se.GetSessionVars()` is not `nil` but got:

```
... Panic: runtime error: invalid memory address or nil pointer dereference (PC=0xB41634)

/usr/lib/go/src/runtime/panic.go:522
  in gopanic
/usr/lib/go/src/runtime/panic.go:82
  in panicmem
/usr/lib/go/src/runtime/signal_unix.go:390
  in sigpanic
session_test.go:2506
  in testSessionSuite.TestGrantViewRelated
/usr/lib/go/src/reflect/value.go:308
  in Value.Call
/home/ejq/go/pkg/mod/github.com/pingcap/check@v0.0.0-20190102082844-67f458068fc8/check.go:836
  in suiteRunner.forkTest.func1
/home/ejq/go/pkg/mod/github.com/pingcap/check@v0.0.0-20190102082844-67f458068fc8/check.go:730
  in suiteRunner.forkCall.func1
/usr/lib/go/src/runtime/asm_amd64.s:1337
  in goexit
```

**Comment by erjiaqing** [comment]
Seems test suite does not set SessionCtx.User, so maybe we cannot change user in unit tests.

**Comment by XuHuaiyu** [comment]
Maybe we can set User for all the testSuite?

**Comment by wjhuang2016** [comment]
@erjiaqing Because the User in testSuite is empty.
Try this:
`tk.Se.GetSessionVars().User = &auth.UserIdentity{Username: "user123", AuthUsername: "auth_user"}`

**Comment by erjiaqing** [comment]
> Maybe we can set User for all the testSuite?

I set default user to root@% and login as root@localhost

Failed on many tests.

**Comment by eurekaka** [comment]
@XuHuaiyu PTAL

**Comment by AndrewDi** [comment]
@erjiaqing You can take privileges_test as example. https://github.com/pingcap/tidb/blob/de113b275476cd087dbf1d8fed926c30c6402ddf/privilege/privileges/privileges_test.go#L279-L300

**Comment by eurekaka** [comment]
@XuHuaiyu PTAL

**Pull request merged**